pyccl/tests/test_tracers.py

```diff
@@ -151,3 +155,90 @@ def test_tracer_nz_support():
 
     with pytest.raises(ValueError):
         _ = ccl.CMBLensingTracer(calculator_cosmo, z_source=2.0)
+
+
+def test_tracer_nz_norm_spline_vs_gsl_intergation():
+    # Create a new Cosmology object so that we're not messing with the other
+    # tests
+    cosmo = ccl.Cosmology(Omega_c=0.27, Omega_b=0.045, h=0.67,
+                          sigma8=0.8, n_s=0.96,
+                          transfer_function='bbks',
+                          matter_power_spectrum='linear')
+    cosmo.cosmo.gsl_params.NZ_NORM_SPLINE_INTEGRATION = True
+    tr_wl, _ = get_tracer("wl", cosmo)
+    tr_nc, _ = get_tracer("nc", cosmo)
+
+    w_wl_spline, _ = tr_wl.get_kernel(chi=None)
+    w_nc_spline, _ = tr_nc.get_kernel(chi=None)
+
+    cosmo.cosmo.gsl_params.NZ_NORM_SPLINE_INTEGRATION = False
+    tr_wl, _ = get_tracer("wl", cosmo)
+    tr_nc, _ = get_tracer("nc", cosmo)
+
+    w_wl_gsl, _ = tr_wl.get_kernel(chi=None)
+    w_nc_gsl, _ = tr_nc.get_kernel(chi=None)
+
+    for w_spline, w_gsl in zip(w_wl_spline, w_wl_gsl):
+        assert np.allclose(w_spline, w_gsl)
+    for w_spline, w_gsl in zip(w_nc_spline, w_nc_gsl):
+        assert np.allclose(w_spline, w_gsl)
+
+
+def test_tracer_lensing_kernel_spline_vs_gsl_intergation():
+    # Create a new Cosmology object so that we're not messing with the other
+    # tests
+    cosmo = ccl.Cosmology(Omega_c=0.27, Omega_b=0.045, h=0.67,
+                          sigma8=0.8, n_s=0.96,
+                          transfer_function='bbks',
+                          matter_power_spectrum='linear')
+    z = np.linspace(0., 1., 2000)
+    n = dndz(z)
+
+    cosmo.cosmo.gsl_params.LENSING_KERNEL_SPLINE_INTEGRATION = True
+    tr_wl = ccl.WeakLensingTracer(cosmo, dndz=(z, n))
+    w_wl_spline, _ = tr_wl.get_kernel(chi=None)
+
+    cosmo.cosmo.gsl_params.LENSING_KERNEL_SPLINE_INTEGRATION = False
+    tr_wl = ccl.WeakLensingTracer(cosmo, dndz=(z, n))
+    w_wl_gsl, _ = tr_wl.get_kernel(chi=None)
+
+    assert np.allclose(w_wl_spline[0], w_wl_gsl[0])
```

Comment:
This has an absolute tolerance of 1E-8. Can you remind me what the order of magnitude is for the lensing kernel?

Reply:
About 1e-5. I've added absolut tolerances now.